command/agent/auth/aws/aws.go

```diff
@@ -183,18 +182,23 @@ func (a *awsMethod) Authenticate(ctx context.Context, client *api.Client) (retTo
 	a.logger.Trace("beginning authentication")
 
 	data := make(map[string]interface{})
+	identityURL := awsutil.InstanceMetadataService.BaseURL + awsutil.InstanceMetadataService.IdentityEndpoint
 
 	switch a.authType {
 	case typeEC2:
 		client := cleanhttp.DefaultClient()
 
 		// Fetch document
 		{
```

Comment:
If you just used [`EC2Metadata.GetDynamicData`](https://docs.aws.amazon.com/sdk-for-go/api/aws/ec2metadata/#EC2InstanceIdentityDocument) would that take care of using IMDSv2 with the proper tokens and such?

Reply:
If I remember, I _think_ this code uses raw HTTP requests because they're passed onto Vault intact. When we used the AWS client, we weren't able to get at that easily. However, I don't want to dig too deep here because re-evaluating the approach taken by the rest of the code expands the scope of this PR.

Reply:
Hi! I'm looking at this again and I think I misunderstood your suggestion. I was bouncing between a lot of things at the time (and still!) 

Hm, that's a good idea. I'll try it out and see how it plays out.

Reply:
That did turn out to be way better, turned out I didn't need a Vault-side EC2Metadata utility at all. Just pushed up the changes after verifying it on a real EC2 instance.